coucharchive

```diff
@@ -281,6 +281,30 @@ def replicate_couchdb_server(source_url, target_url, ignore_dbs=[]):
         pool.join()
 
 
+def create_missing_documents(source_db, target_db):
+    # Manually create documents on the target database that where previously
+    # deleted and re-created on the source. It avoids CouchDB replication bug.
+    # See: https://github.com/apache/couchdb/issues/1418
+
+    missing_documents = [doc for doc in source_db if doc not in target_db]
+
+    for doc_id in missing_documents:
+        source_doc = source_db[doc_id]
+
+        # Not needed. Will be generated on target.
+        # Plus avoids conflicts.
+        del source_doc['_rev']
+
+        try:
+            _, target_rev = target_db.save(source_doc)
+        except couchdb.http.HTTPError:
+            raise
+
+        if target_rev != source_doc.rev:
+            raise Exception('%s: source and target documents have diverged'
+                            % source_db.name)
```

Comment:
A suggestion to make this more user-friendly and understandable in logs:
```suggestion
            raise Exception('bug_1418_create_missing_documents: '
                            '%s: source and target documents have diverged'
                            % source_db.name)
```

Reply:
Agreed! :+1: